default/data_file.py

```diff
@@ -91,7 +92,7 @@ def write_json_to_zip(self, files):
         if isinstance(files, dict):
             files = files.items()
         for name, content in files:
-            with ZipFile(self.path, 'w', compression=ZIP_DEFLATED) as zipfile:
+            with ZipFile(self.path, 'a', compression=ZIP_DEFLATED) as zipfile:
```

Comment:
Why was this changed to append mode?

Reply:
Because the Upgrade feature expects to return all files in a single zip. When using `w` only the last file is returned.

Reply:
I think if we swap the position of the for-loop and with-statement, then we can use regular 'w' mode.

Reply:
ok! I'll try.